### ISSUE TITLE
Wrapping an rdf:List loses information in pretty-xml serialization

### ISSUE THREAD

**Issue by gjhiggins** [RDFLib/rdflib#160]
azarot...@gmail.com, 2011-06-22T17:10:07.000Z
#### What steps will reproduce the problem?

``` python
from rdflib import ConjunctiveGraph
from rdflib import Namespace, Literal
from rdflib.collection import Collection

# Need to generate the following:

foo = Namespace('http://www.example.org/foo/ns/')
ex = Namespace('http://www.example.org/example/foo/')
rdf = Namespace('http://www.w3.org/1999/02/22-rdf-syntax-ns#')

# Works:  x a rdf:List, a foo:Other ;
# Fails:  y a foo:Wrapper, foo:wraps x; x a rdf:List, a foo:Other ;

g = ConjunctiveGraph()

bits = [ex['a'], ex['b'], ex['c']]
l = Collection(g, ex['thing'], bits)
g.add((ex['thing'], rdf['type'], foo['Other']))
g.add((ex['thing'], foo['property'], Literal('Some Value')))
for b in bits:
    g.add((b, rdf['type'], foo['Item']))

print "Works:"
print g.serialize(format='pretty-xml')

g.add((ex['wrapper'], rdf['type'], foo['Wrapper']))
g.add((ex['wrapper'], foo['wraps'], ex['thing']))

print ""
print "Fails (info is lost):"
print g.serialize(format='pretty-xml')
```
#### What is the expected output? What do you see instead?

Equivalent of n3, or regular xml.
#### What version of the product are you using? On what operating system?

3.1, MacOSX

**Comment by kouralex** [comment]
Sorry for commenting to an old issue, but neither do I understand why some assertions are ignored, can't they just be taken into account?

Using Apache Jena's `rdfdiff` I can confirm that models produced by vanilla XML serializer and Turtle serializer are equal but the one generated by `PrettyXmlSerializer` is not.

**Comment by aucampia** [comment]
@kouralex best to open an issue, but if something is wrong with RDFLib we are happy to have it fixed, but RDFLib is entirely community-driven.

**Comment by kouralex** [comment]
@aucampia yes, and as I have maintained an open source software, I understand. I was just wondering if there are some specification that let one to omit some declarations/assertions but that does not seem to be the case - hence the comment. I am not overly into/interested in the (RDF/)XML format(s), in my use case I am basically just offering different serializations of the graph to the user. Therefore, I can use the plain XML version (or use another program to generate RDF/XML-ABBREV) if I want to, so this indeed is not an issue to me.

**Comment by aucampia** [comment]
@kouralex if a triple is in asserted in the input source it should be in the output, so if it is ignored it is a bug. I actually don't think this should have been closed, as it seems the last communication on this made it clear it is still a problem.
